### PR TITLE
Change include to support all forms.

### PR DESCRIPTION
**Pull request by sparhami**
Also throw an error with the final state before timing out.

### CODE REVIEW

File: spec/test-expect.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ describes.endtoend('AMP carousel', {
     await expect(2+2).to.be.above(3);
     await expect(2+2).to.be.below(5);
 
+    await expect({a: '2', b: '3'}).to.include({a: '2'});
```
Comment (cvializ) [inline spec/test-expect.js:22]: Can you add a similar test that uses a `ControllerPromise` result? Like the tests below in `should work with async values`

File: lib/expect.js [hunk 2]
```diff
@@ -127,6 +139,42 @@ function overwriteWithCondition(utils, condition) {
...
+        return _super.apply(this, arguments);
+      }
+
+      const resultPromise = waitForValue((obj) => {
+        try {
+          flag(this, 'object', obj);
+          _super.apply(this, arguments);
+          return true;
+        } catch(e) {
+          lastExpectError = e;
```
Comment (cvializ) [inline lib/expect.js:170]: What might cause `_super.apply` to throw? Is the return value `true` or `false` significant to anywhere downstream in the Promise chain?

File: spec/test-expect.js (hunk 1 above)
Comment (sparhami) [reply spec/test-expect.js:22]: Done. I can add a more realistic test later that does something that makes more sense (verifying a client rect). I didn't want to mix those changes into this PR, but I can add that later.

File: lib/expect.js (hunk 2 above)
Comment (sparhami) [reply lib/expect.js:170]: Added some comments to explain what the code is doing.

**Comment by cvializ** [comment]
Nice thanks!

**Pull request merged**